pkg/metrics/httpmetricmiddleware.go

```diff
@@ -0,0 +1,80 @@
+package metrics
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/kubecost/events"
+)
+
+// ResponseMetricMiddleware dispatches metric events for handles request and responses.
+func ResponseMetricMiddleware(handler http.Handler) http.Handler {
+	dispatcher := events.GlobalDispatcherFor[HttpHandlerMetricEvent]()
```

Comment:
Should I be using `GlobalDispatcherFor` in ETL, do you think?

Reply:
This is where my code review for you _really_ becomes your code review for me :laughing:

Reply:
I think it just depends -- I really don't care much for global singleton patterns, but I also don't like having to propagate the depths of the architecture to wire up eventing. 

It made sense to me to allow localized dispatchers for systems like ETL, but I did want to "normalize" a single typed event dispatcher globally as well. In the case of metrics and the http middleware, these things are already "global" in nature, so appending the global dispatching was an easy decision. For ETL, it's really whether or not we want the events to be "exposed" - It's also perfectly reasonable to have a requirement like: If you want to subscribe to ETL events, you need to have a reference to the dispatcher created inside the ETL types. (Versus just allowing any part of the code to grab the global dispatcher and start spying 😄 )

Reply:
Switched to using the global. I like it more.